### PR TITLE
VTCCA-4720 added aria-lable

### PR DESCRIPTION
**Pull request by zahiriqbal**
https://jira.tools.tax.service.gov.uk/browse/VTCCA-4720

### CODE REVIEW

**Comment by darrell-plant** [comment]
code duplication needs fixing 
then hanging " to " needs fixing
logic to determine Current|Draft|Historic is not right
also, odd formattting

File: app/views/dashboard/assessmentDetails.scala.html [hunk 1]
```diff
@@ -33,10 +32,36 @@
...
+        (assessment.currentFromDate, assessment.currentToDate) match {
+            case (Some(_), None)     => Some(Messages("assessments.enddate.present"))
+            case (Some(_), Some(to)) => Some(Formatters.formatDate(to))
+            case _                   => None
+        }
+    } else None
+}
+
+@ariaLabel(assessment: ApiAssessment, listType: String, isDraft: Boolean) = @{
+    val listTypeText = listType.takeWhile(_ != 'A').capitalize
```
Comment (PeterPerhac) [inline app/views/dashboard/assessmentDetails.scala.html:54]: what does this line do?

Comment (PeterPerhac) [reply app/views/dashboard/assessmentDetails.scala.html:54]: `listType.takeWhile(_ != 'A').capitalize`

File: app/views/dashboard/assessmentDetails.scala.html [hunk 2]
```diff
@@ -33,10 +32,36 @@
...
+    val endText = endDate(assessment, isDraft).map(ed => s"to $ed").getOrElse("").trim
+
+    s"Select this valuation - $listTypeText $fromText $endText"
+}
+
+@table(list: Seq[(String, ApiAssessment)], listType: String, isDraft: Boolean) = {
+    <div id="viewAssessmentRadioGroup_@listType" class="form-group">
         <div class="table-wrapper-no-pad">
-            <table id="@tableId" class="margin-bottom-10 margin-top-10">
+            <table id="@listType" class="margin-bottom-10 margin-top-10">
```
Comment (PeterPerhac) [inline app/views/dashboard/assessmentDetails.scala.html:64]: what is `listType` ? is it safe to use this String as an HTML ID attribute?

File: app/views/dashboard/assessmentDetails.scala.html [hunk 3]
```diff
@@ -120,13 +140,13 @@
 <div id="details-content-valuations" class="govuk-body">
 
     @if(model.currentAssessments.nonEmpty){
-    <h2 class="govuk-heading-m">@messages("assessments.current.title")</h2>
-        @table(model.currentAssessments, "currentAssessments")
+        <h2 class="govuk-heading-m">@messages("assessments.current.title")</h2>
+        @table(model.currentAssessments, listType = "currentAssessments", isDraft = false)
```
Comment (PeterPerhac) [inline app/views/dashboard/assessmentDetails.scala.html:144]: I don't understand where does this "currentAssessments" come from? That's not a list type...

File: app/views/dashboard/assessmentDetails.scala.html [hunk 4]
```diff
@@ -57,53 +82,48 @@
...
                     @list.map {  case (url, assessment) =>
-                        <tr class="govuk-table__row">
-                            <td class="govuk-table__cell">
-                                @assessment.effectiveDate.map(Formatters.formatDate).getOrElse("")
-                            </td>
+                    <tr class="govuk-table__row">
+
+                        <td class="govuk-table__cell">@effectiveDate(assessment)</td>
+
+                        @assessment.rateableValue match {
```
Comment (PeterPerhac) [inline app/views/dashboard/assessmentDetails.scala.html:89]: un thus template I see the same kind of problem repeatedly:

```
 @assessment.rateableValue match {
                            case Some(rv) => {
                                <td class="govuk-table__cell">£@rv</td>
                            }
                            case None => {
                                <td class="govuk-table__cell">@messages("assessments.na")</td>
                            }
                        }
```

regardless of whether the value is Some or None, you'd always want to output the  `<td class="govuk-table__cell">` and the `</td>`

Why repeat the element?

this sort of code is what you want I think:

```
<td class="govuk-table__cell">
  @assessment.rateableValue.fold(messages("assessments.na"))( rv => s"£$rv" )
</td>
```

File: app/views/dashboard/assessmentDetails.scala.html (hunk 2 above)
Comment (darrell-plant) [reply app/views/dashboard/assessmentDetails.scala.html:64]: yes - and it hasn't changed - just the name.

File: app/views/dashboard/assessmentDetails.scala.html (hunk 1 above)
Comment (darrell-plant) [reply app/views/dashboard/assessmentDetails.scala.html:54]: Splits the input "currentAssessments" (for example) to "Current"

File: app/views/dashboard/assessmentDetails.scala.html (hunk 3 above)
Comment (darrell-plant) [reply app/views/dashboard/assessmentDetails.scala.html:144]: explained offline

File: app/views/dashboard/assessmentDetails.scala.html (hunk 4 above)
Comment (darrell-plant) [reply app/views/dashboard/assessmentDetails.scala.html:89]: true - it's existing code, but I will tidy that up

File: app/views/dashboard/assessmentDetails.scala.html (hunk 3 above)
Comment (darrell-plant) [reply app/views/dashboard/assessmentDetails.scala.html:144]: agreed approach

**Comment by zahiriqbal** [comment]
👍

**Pull request merged**